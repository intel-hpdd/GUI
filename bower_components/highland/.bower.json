{
  "name": "highland",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "version": "2.7.1",
  "_release": "2.7.1",
  "_resolution": {
    "type": "version",
    "tag": "2.7.1",
    "commit": "197822ee483f1e38fd410863d0edb4ad07d06d4d"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.7.1",
  "_originalSource": "highland"
}