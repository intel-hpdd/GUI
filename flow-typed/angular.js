// @flow

declare module angular {
  declare class $scope {
    $new():$scope;
    $digest():void;
  }
  declare type $scopeT = $scope;
  declare class Module {
    provider(name:string, providerType:Function):Module;
    factory(name:string, providerFunction:Function):Module;
    controller(name:string, constructor:Function):Module;
    service(name:string, constructor:Function):Module;
    config(configFn:Function):Module;
    run(initializationFn:Function):Module;
    directive(name:string, directiveFactory:Function):Module;
    component(name:string, options:Object):Module;
    filter(name:string, filterFactory:Function):Module;
    value(name:string, object:any):Module;
    constant(name:string, object:any):Module;
    name:string;
  }
  declare type $compileT = (el:(string | HTMLElement)) => (scope:$scope) => HTMLElement[];
  declare module.exports: {
    module(name:string, dependencies: string[]):Module;
    bootstrap(element:(HTMLElement | Document), modules:string[], config:Object):void;
    merge(...rest:Object[]):Object;
    element(el:(string | HTMLElement)): {
      clone: () => HTMLElement
    };
  }
}
