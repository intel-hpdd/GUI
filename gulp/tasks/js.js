//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
