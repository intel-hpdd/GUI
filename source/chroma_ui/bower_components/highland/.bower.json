{
  "name": "highland",
  "version": "2.4.0",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "_release": "2.4.0",
  "_resolution": {
    "type": "version",
    "tag": "2.4.0",
    "commit": "7ff0deb4abec773009324072d8855f394cc93467"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.4.0",
  "_originalSource": "highland"
}