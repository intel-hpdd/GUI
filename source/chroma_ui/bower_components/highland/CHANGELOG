Changelog
=========

This file does not aim to be comprehensive (you have git history for that),
rather it lists changes that might impact your own code as a consumer of
this library.

2.0.0
-----

* The `source.merge()` algorithm now evaluates the entire source stream before
  reading from all of the resulting streams in parallel (previously it would
  start reading as soon as the source emitted the next stream)
* The `merge()` function now attempts to balance inputs more fairly. For example,
  if stream A has 100 values buffered and stream B gets a new value after 100ms,
  if we read at 200ms we'll get a value from each stream. Previously it would
  exhaust the stream A buffer before reading from stream B.
