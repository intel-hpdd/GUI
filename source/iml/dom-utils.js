// @flow

type QuerySelector<T: HTMLElement> = (HTMLElement | Document, string) => T;

export const querySelector: QuerySelector<*> = (
  elm: HTMLElement | Document,
  sel: string
) => {
  const selected = elm.querySelector(sel);

  if (!selected)
    throw new Error(
      `Could not find selector ${sel} for element ${elm.nodeName}`
    );

  return selected;
};
