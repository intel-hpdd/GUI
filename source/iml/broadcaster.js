// @flow

//
// Copyright (c) 2017 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import highland from 'highland';

import type {
  HighlandStreamT,
  errorWrapT
} from 'highland';


type streamFnT = () => HighlandStreamT<any>;

export default function broadcaster (source$:HighlandStreamT<any>):streamFnT {
  var latest:mixed;

  const viewers = [];

  source$
    .errors(error => {
      const err:errorWrapT = {
        __HighlandStreamError__: true,
        error
      };

      viewers.forEach(v => v.write(err));
    })
    .each((xs:mixed) => {
      latest = xs;

      viewers.forEach(v => v.write(xs));
    });

  const fn = () => {
    const viewer$:HighlandStreamT<any> = highland()
      .onDestroy(() => {
        const idx = viewers
          .indexOf(viewer$);

        if (idx !== -1)
          viewers.splice(idx, 1);
      });

    if (latest)
      viewer$.write(latest);

    viewers.push(viewer$);

    return viewer$;
  };

  fn.endBroadcast = () => source$.destroy();

  return fn;
}
