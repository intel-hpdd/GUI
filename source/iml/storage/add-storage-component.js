// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import type { Field, StorageResourceClass } from './storage-types.js';
import type { State } from './storage-reducer.js';

import Inferno from 'inferno';
import Component from 'inferno-component';

import { asViewer } from '../as-viewer/as-viewer';
import TableFilter from './table-filter.js';
import socketStream from '../socket/socket-stream.js';
import { UI_ROOT } from '../environment.js';

const PageAlert = ({ children, className }) => {
  if (!children) return;

  return <div className={`alert span4 offset4 ${className}`}>{children}</div>;
};

const FormItem = ({ field, value, onInput }: { field: Field, value: string, onInput: Function }) => (
  <div class="form-group">
    <label for={field.name}>{field.label}</label>
    <input
      required={!field.optional}
      value={value}
      onInput={onInput}
      type={field.class === 'Password' ? 'password' : 'text'}
      class="form-control"
      id={field.name}
    />
  </div>
);

type AddFormState = {
  form: {
    [number]: {
      [string]: string
    }
  },
  loading: boolean,
  saveFailed: ?boolean
};

type AddFormProps = {
  resourceClass: StorageResourceClass
};

class AddStorageForm extends Component {
  state: AddFormState;
  props: AddFormProps;
  constructor(props: AddFormProps) {
    super(props);

    this.state = {
      form: {
        [props.resourceClass.id]: {}
      },
      loading: false,
      saveFailed: null
    };
  }
  onInput(id: number, name: string, ev: SyntheticInputEvent) {
    this.setState({
      form: {
        ...this.state.form,
        [id]: {
          ...this.state.form[id],
          [name]: ev.target.value
        }
      }
    });
  }
  handleSubmit(id: number, ev: Event) {
    Event.prototype.preventDefault.call(ev);
    Event.prototype.stopImmediatePropagation.call(ev);
    this.setState({ loading: true, saveFailed: null });

    socketStream(
      '/storage_resource',
      {
        method: 'post',
        json: {
          attrs: this.state.form[id],
          plugin_name: this.props.resourceClass.plugin_name,
          class_name: this.props.resourceClass.class_name
        }
      },
      true
    ).pull(err => {
      this.setState({
        loading: false,
        saveFailed: err ? true : false,
        form: {
          ...this.state.form,
          [id]: err ? this.state.form[id] : {}
        }
      });
    });
  }
  componentWillReceiveProps({ resourceClass }) {
    if (this.state.form[resourceClass.id] == null)
      this.setState({
        loading: false,
        saveFailed: null,
        form: {
          ...this.state.form,
          [resourceClass.id]: {}
        }
      });
  }

  render() {
    const id = this.props.resourceClass.id;

    return (
      <form autocomplete="off" onSubmit={this.handleSubmit.bind(this, id)}>
        <PageAlert className={this.state.saveFailed ? 'alert-danger' : 'alert-success'}>
          {(() => {
            switch (this.state.saveFailed) {
              case true:
                return 'An unexpected error occured. Please update form and resubmit.';
              case false:
                return 'Record Created.';
              default:
                return null;
            }
          })()}
        </PageAlert>
        {this.props.resourceClass.fields.map(x => (
          <FormItem
            field={x}
            key={x.name}
            value={this.state.form[id][x.name]}
            onInput={this.onInput.bind(this, id, x.name)}
          />
        ))}
        <button disabled={this.state.loading} type="submit" class="btn btn-success">
          Save
          <i className={`fa ${this.state.loading ? 'fa-spinner fa-spin' : 'fa-check-circle'}`} />
        </button>
      </form>
    );
  }
}

export const AddStorageComponent = asViewer(
  'storage',
  ({
    storage: {
      resourceClasses,
      config: { selectIndex }
    }
  }: {
    storage: State
  }) => {
    if (!resourceClasses || selectIndex == null || !resourceClasses[selectIndex]) return;

    return (
      <div class="container container-full storage container">
        <h4 class="section-header">Add Storage Device</h4>
        <TableFilter classes={resourceClasses} idx={selectIndex} />
        <AddStorageForm resourceClass={resourceClasses[selectIndex]} />
        <div style={{ margin: '25px 0' }}>
          <a href={`${UI_ROOT}configure/storage`}>&larr; Return to storage configuration page</a>
        </div>
      </div>
    );
  }
);

export default {
  bindings: { storageB: '<' },
  controller: function($element: HTMLElement[]) {
    'ngInject';

    const el = $element[0];

    this.$onInit = () => {
      Inferno.render(<AddStorageComponent viewer={this.storageB} />, el);
    };

    this.$onDestroy = () => {
      Inferno.render(null, el);

      this.storageB.endBroadcast();
    };
  }
};
