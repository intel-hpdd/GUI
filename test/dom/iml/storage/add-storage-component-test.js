// @flow

import angular from "../../../angular-mock-setup.js";
import highland, { type HighlandStreamT } from "highland";
import broadcaster from "../../../../source/iml/broadcaster.js";
import type { State } from "../../../../source/iml/storage/storage-reducer.js";
import { querySelector } from "../../../../source/iml/dom-utils.js";

describe("storage component", () => {
  let mockSocketStream, storage$: HighlandStreamT<State>, $scope, $compile, template, root;

  beforeEach(
    angular.mock.module($compileProvider => {
      mockSocketStream = jest.fn();
      jest.mock("../../../../source/iml/socket/socket-stream.js", () => mockSocketStream);

      $compileProvider.component(
        "addStorage",
        require("../../../../source/iml/storage/add-storage-component.js").default
      );

      storage$ = highland();
    })
  );

  beforeEach(
    angular.mock.inject((_$compile_, $rootScope) => {
      $scope = $rootScope.$new();
      $compile = _$compile_;
    })
  );

  beforeEach(() => {
    $scope.storage$ = broadcaster(storage$);
    template = '<add-storage storage-b="storage$"></add-storage>';
  });

  it("should render nothing with no data", () => {
    storage$.write({
      config: {
        sortDesc: false,
        sortKey: "name",
        loading: false,
        selectIndex: 0,
        entries: 10,
        offset: 0
      },
      resourceClasses: [],
      resources: {
        meta: {
          limit: 10,
          next: null,
          offset: 0,
          previous: null,
          total_count: 0
        },
        objects: []
      }
    });

    root = $compile(template)($scope)[0];

    expect(root).toMatchSnapshot();
  });

  it("should render with data", () => {
    storage$.write({
      config: {
        sortDesc: false,
        sortKey: "name",
        loading: true,
        selectIndex: 0,
        entries: 1,
        offset: 0
      },
      resourceClasses: [
        {
          class_name: "EMCPower",
          columns: [
            {
              label: "Size",
              name: "size"
            },
            {
              label: "Filesystem type",
              name: "filesystem_type"
            },
            {
              label: "Uuid",
              name: "uuid"
            }
          ],
          fields: [
            {
              class: "Bytes",
              label: "Size",
              name: "size",
              optional: false,
              user_read_only: false
            },
            {
              class: "Boolean",
              label: "Filesystem type",
              name: "filesystem_type",
              optional: true,
              user_read_only: false
            },
            {
              class: "String",
              label: "Uuid",
              name: "uuid",
              optional: false,
              user_read_only: false
            }
          ],
          id: 1,
          label: "linux-EMCPower",
          modified_at: "2017-07-19T13:58:02.615410",
          plugin_internal: true,
          plugin_name: "linux",
          resource_uri: "/api/storage_resource_class/1/",
          user_creatable: false
        }
      ],
      resources: {
        meta: {
          limit: 10,
          next: null,
          offset: 0,
          previous: null,
          total_count: 1
        },
        objects: []
      }
    });

    root = $compile(template)($scope)[0];

    expect(root).toMatchSnapshot();
  });

  describe("interaction", () => {
    let stream;

    beforeEach(() => {
      stream = highland();
      mockSocketStream.mockReturnValue(stream);

      storage$.write({
        config: {
          sortDesc: false,
          sortKey: "name",
          loading: true,
          selectIndex: 0,
          entries: 1,
          offset: 0
        },
        resourceClasses: [
          {
            class_name: "EMCPower",
            columns: [
              {
                label: "Size",
                name: "size"
              },
              {
                label: "Filesystem type",
                name: "filesystem_type"
              },
              {
                label: "Uuid",
                name: "uuid"
              }
            ],
            fields: [
              {
                class: "Bytes",
                label: "Size",
                name: "size",
                optional: false,
                user_read_only: false
              },
              {
                class: "Boolean",
                label: "Filesystem type",
                name: "filesystem_type",
                optional: true,
                user_read_only: false
              },
              {
                class: "String",
                label: "Uuid",
                name: "uuid",
                optional: false,
                user_read_only: false
              }
            ],
            id: 1,
            label: "linux-EMCPower",
            modified_at: "2017-07-19T13:58:02.615410",
            plugin_internal: true,
            plugin_name: "linux",
            resource_uri: "/api/storage_resource_class/1/",
            user_creatable: false
          }
        ],
        resources: {
          meta: {
            limit: 10,
            next: null,
            offset: 0,
            previous: null,
            total_count: 1
          },
          objects: []
        }
      });

      root = $compile(template)($scope)[0];
      querySelector(document, "body").appendChild(root);
    });

    afterEach(() => {
      querySelector(document, "body").removeChild(root);
    });

    describe("submitting the form", () => {
      beforeEach(() => {
        const size: HTMLInputElement = (querySelector(root, "#size"): any);
        size.value = "foo";
        size.dispatchEvent(new Event("input"));

        const uuid: HTMLInputElement = (querySelector(root, "#uuid"): any);
        uuid.value = "bar";
        uuid.dispatchEvent(new Event("input"));

        querySelector(root, "button").dispatchEvent(new MouseEvent("click"));
      });

      it("should submit the form", () => {
        expect(mockSocketStream).toHaveBeenCalledWith(
          "/storage_resource",
          {
            json: {
              attrs: { size: "foo", uuid: "bar" },
              class_name: "EMCPower",
              plugin_name: "linux"
            },
            method: "post"
          },
          true
        );
      });

      it("should render loading", () => {
        expect(root).toMatchSnapshot();
      });

      it("should render success", () => {
        stream.write({});

        expect(root).toMatchSnapshot();
      });

      it("should render error", () => {
        // $FlowFixMe: Internal error type for testing
        stream.write({
          __HighlandStreamError__: true,
          error: new Error("boom")
        });

        expect(root).toMatchSnapshot();
      });
    });
  });
});
