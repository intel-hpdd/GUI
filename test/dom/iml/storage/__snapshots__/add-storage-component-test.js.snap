// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storage component interaction submitting the form should render error 1`] = `
<div>
  <div
    class="container container-full storage container"
  >
    <h4
      class="section-header"
    >
      Add Storage Device
    </h4>
    <div
      class="form-group well"
    >
      <label
        class="control-label"
      >
        Select Resource Class
      </label>
      <select
        class="form-control"
      >
        <option
          value="EMCPower,linux"
        >
          linux-EMCPower
        </option>
      </select>
    </div>
    <form
      autocomplete="off"
    >
      <div
        class="alert span4 offset4 alert-danger"
      >
        An unexpected error occured. Please update form and resubmit.
      </div>
      <div
        class="form-group"
      >
        <label
          for="size"
        >
          Size
        </label>
        <input
          class="form-control"
          id="size"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          for="filesystem_type"
        >
          Filesystem type
        </label>
        <input
          class="form-control"
          id="filesystem_type"
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          for="uuid"
        >
          Uuid
        </label>
        <input
          class="form-control"
          id="uuid"
          required=""
          type="text"
        />
      </div>
      <button
        class="btn btn-success"
        type="submit"
      >
        Save
        <i
          class="fa fa-check-circle"
        />
      </button>
    </form>
    <div
      style="margin: 25px 0px;"
    >
      <a
        href="http://localhost/configure/storage"
      >
        ← Return to storage configuration page
      </a>
    </div>
  </div>
</div>
`;

exports[`storage component interaction submitting the form should render loading 1`] = `
<div>
  <div
    class="container container-full storage container"
  >
    <h4
      class="section-header"
    >
      Add Storage Device
    </h4>
    <div
      class="form-group well"
    >
      <label
        class="control-label"
      >
        Select Resource Class
      </label>
      <select
        class="form-control"
      >
        <option
          value="EMCPower,linux"
        >
          linux-EMCPower
        </option>
      </select>
    </div>
    <form
      autocomplete="off"
    >
      
      <div
        class="form-group"
      >
        <label
          for="size"
        >
          Size
        </label>
        <input
          class="form-control"
          id="size"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          for="filesystem_type"
        >
          Filesystem type
        </label>
        <input
          class="form-control"
          id="filesystem_type"
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          for="uuid"
        >
          Uuid
        </label>
        <input
          class="form-control"
          id="uuid"
          required=""
          type="text"
        />
      </div>
      <button
        class="btn btn-success"
        disabled=""
        type="submit"
      >
        Save
        <i
          class="fa fa-spinner fa-spin"
        />
      </button>
    </form>
    <div
      style="margin: 25px 0px;"
    >
      <a
        href="http://localhost/configure/storage"
      >
        ← Return to storage configuration page
      </a>
    </div>
  </div>
</div>
`;

exports[`storage component interaction submitting the form should render success 1`] = `
<div>
  <div
    class="container container-full storage container"
  >
    <h4
      class="section-header"
    >
      Add Storage Device
    </h4>
    <div
      class="form-group well"
    >
      <label
        class="control-label"
      >
        Select Resource Class
      </label>
      <select
        class="form-control"
      >
        <option
          value="EMCPower,linux"
        >
          linux-EMCPower
        </option>
      </select>
    </div>
    <form
      autocomplete="off"
    >
      <div
        class="alert span4 offset4 alert-success"
      >
        Record Created.
      </div>
      <div
        class="form-group"
      >
        <label
          for="size"
        >
          Size
        </label>
        <input
          class="form-control"
          id="size"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          for="filesystem_type"
        >
          Filesystem type
        </label>
        <input
          class="form-control"
          id="filesystem_type"
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          for="uuid"
        >
          Uuid
        </label>
        <input
          class="form-control"
          id="uuid"
          required=""
          type="text"
        />
      </div>
      <button
        class="btn btn-success"
        type="submit"
      >
        Save
        <i
          class="fa fa-check-circle"
        />
      </button>
    </form>
    <div
      style="margin: 25px 0px;"
    >
      <a
        href="http://localhost/configure/storage"
      >
        ← Return to storage configuration page
      </a>
    </div>
  </div>
</div>
`;

exports[`storage component should render nothing with no data 1`] = `
<div>
  
</div>
`;

exports[`storage component should render with data 1`] = `
<div>
  <div
    class="container container-full storage container"
  >
    <h4
      class="section-header"
    >
      Add Storage Device
    </h4>
    <div
      class="form-group well"
    >
      <label
        class="control-label"
      >
        Select Resource Class
      </label>
      <select
        class="form-control"
      >
        <option
          value="EMCPower,linux"
        >
          linux-EMCPower
        </option>
      </select>
    </div>
    <form
      autocomplete="off"
    >
      
      <div
        class="form-group"
      >
        <label
          for="size"
        >
          Size
        </label>
        <input
          class="form-control"
          id="size"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          for="filesystem_type"
        >
          Filesystem type
        </label>
        <input
          class="form-control"
          id="filesystem_type"
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          for="uuid"
        >
          Uuid
        </label>
        <input
          class="form-control"
          id="uuid"
          required=""
          type="text"
        />
      </div>
      <button
        class="btn btn-success"
        type="submit"
      >
        Save
        <i
          class="fa fa-check-circle"
        />
      </button>
    </form>
    <div
      style="margin: 25px 0px;"
    >
      <a
        href="http://localhost/configure/storage"
      >
        ← Return to storage configuration page
      </a>
    </div>
  </div>
</div>
`;
